hr_job_categories/models/hr.py

```diff
@@ -0,0 +1,86 @@
+# Copyright (C) 2021 Trevi Software (https://trevi.et)
+# Copyright (C) 2013 Michael Telahun Makonnen <mmakonnen@gmail.com>.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class HrJob(models.Model):
+
+    _inherit = "hr.job"
+
+    category_ids = fields.Many2many(
+        string="Associated Tags",
+        comodel_name="hr.employee.category",
+        relation="job_category_rel",
+        column1="job_id",
+        column2="category_id",
+    )
+
+
+class HrContract(models.Model):
+
+    _name = "hr.contract"
+    _inherit = "hr.contract"
+
+    @api.model
+    def _remove_tags(self, employee_id, job_id):
+
+        if not employee_id or not job_id:
+            return
+
+        ee_obj = self.env["hr.employee"]
+        eedata = ee_obj.browse(employee_id).read(["category_ids"])
+        job = self.env["hr.job"].browse(job_id)
+        for tag in job.category_ids:
+            if tag.id in eedata[0]["category_ids"]:
+                ee_obj.browse(employee_id).write({"category_ids": [(3, tag.id)]})
+        return
+
+    @api.model
+    def _tag_employees(self, employee_id, job_id):
+
+        if not employee_id or not job_id:
+            return
+
+        ee_obj = self.env["hr.employee"]
+        eedata = ee_obj.browse(employee_id).read(["category_ids"])
+        job = self.env["hr.job"].browse(job_id)
+        for tag in job.category_ids:
+            if tag.id not in eedata[0]["category_ids"]:
+                ee_obj.browse(employee_id).write({"category_ids": [(4, tag.id)]})
```

Comment:
We don't really need read() here. This whole block can be simplified:
```py
ee = self.env["hr.employee].browse(employee_id)`
job = self.env["hr.job"].browse(job_id)
for tag in job.category_ids:
    if tag.id not in ee.caategory_ids.ids:
        ee.write({"category_ids": [(4, tag.id)]})
```

Reply:
The test is failing because the test case is trying to create a category with the name 'Sales' which is the same category loaded by the demo data for 'hr.employee.category' model, I was running mine without the demo data